web/client/components/TOC/DefaultLayer.jsx

```diff
@@ -164,8 +168,12 @@ class DefaultLayer extends React.Component {
                 : null);
     }
     renderNode = (grab, hide, selected, error, warning, isDummy, other) => {
-        const isEmpty = this.props.node.type === 'wms' && !this.props.activateLegendTool && !this.props.showFullTitleOnExpand
-        || this.props.node.type !== 'wms' && !this.props.showFullTitleOnExpand;
+        const isEmpty = !(
+            this.props.showFullTitleOnExpand
+            || this.props.activateLegendTool
+                && this.props.node.type === 'wms'
+                || ['wfs', 'vector'].includes(this.props.node.type) && this.props.node.style.format === 'geostyler'
```

Comment:
following the previous comment I would propose to remove the `this.props.node.style.format === 'geostyler'` check and include `3dtiles` type. Please do a test with these maps `#/viewer/cesium/39079` (batched 3d model) and `#/viewer/cesium/39241` (point cloud)
```
 || ['wfs', 'vector', '3dtiles'].includes(this.props.node.type)
```

Reply:
We need to check the format here to hide collapsible button (.toc-legend)

Reply:
ok understood, we can keep `this.props.node?.style?.format === 'geostyler'` here but I would pass the style object to the Legend component anyway

Reply:
as discussed we can skip the `3dtiles` for the moment,
@belom88 thanks for checking